executor/src/lib.rs

```diff
@@ -55,14 +51,18 @@ extern crate error_chain;
 #[cfg(test)]
 extern crate assert_matches;
 
+// TODO: move into own crate
+macro_rules! map {
```

Comment:
something that might belong in the `rstd` along with `try_opt`

Reply:
yeah - out of scope for this PR but worth doing soonish.